crates/bevy_core_pipeline/src/upscaling/mod.rs

```diff
@@ -46,28 +46,32 @@ fn prepare_view_upscaling_pipelines(
     let mut output_textures = HashSet::new();
     for (entity, view_target, camera) in view_targets.iter() {
         let out_texture_id = view_target.out_texture().id();
-        let blend_state = if let Some(ExtractedCamera {
-            output_mode: CameraOutputMode::Write { blend_state, .. },
-            ..
-        }) = camera
-        {
-            match *blend_state {
-                None => {
-                    // If we've already seen this output for a camera and it doesn't have a output blend
-                    // mode configured, default to alpha blend so that we don't accidentally overwrite
-                    // the output texture
-                    if output_textures.contains(&out_texture_id) {
-                        Some(BlendState::ALPHA_BLENDING)
-                    } else {
-                        None
+        let blend_state = if let Some(extracted_camera) = camera {
+            match extracted_camera.output_mode {
+                CameraOutputMode::Skip => None,
+                CameraOutputMode::Write { blend_state, .. } => {
+                    let already_seen = output_textures.contains(&out_texture_id);
+                    output_textures.insert(out_texture_id);
+
+                    match blend_state {
+                        None => {
+                            // If we've already seen this output for a camera and it doesn't have a output blend
+                            // mode configured, default to alpha blend so that we don't accidentally overwrite
+                            // the output texture
+                            if already_seen {
+                                Some(BlendState::ALPHA_BLENDING)
+                            } else {
+                                None
+                            }
+                        }
+                        _ => blend_state,
                     }
                 }
-                _ => *blend_state,
             }
         } else {
+            output_textures.insert(out_texture_id);
```

Comment:
I think the idea here was to configure a sensible default blend mode in the event we were writing on top of the same output texture in the csae of `CameraOutputMode::Write { blend_state: None, .. }`. If I'm working through the logic correctly, moving this check would mean we'd only configure alpha blending in the following scenario:
- Camera 1: `CameraOutputMode::Skip`
- Camera 2: `CameraOutputMode::Write { blend_state: None }`

Reply:
No, in that scenario alpha blending wouldn't be configured, because we do not insert the `out_texture_id` into `output_textures` for Camera 1, because it has `CameraOutputMode::Skip`, so `already_seen` would be false for Camera 2.

Alpha blending is configured for a camera A if it's `blend_state` is None and we've iterated over a `ViewTarget` that is NOT a camera with `CameraOutputMode::Skip` and has the same output texture as camera A. So for example:
- Camera 1: CameraOutputMode::Write { blend_state: ... }
- Camera 2: CameraOutputMode::Write { blend_state: None }

where both cameras have the same output texture would lead to Camera 2 getting a `blend_state` of alpha blending.